File: app/src/main/java/com/mainstreetcode/teammate/model/enums/BlockReason.java
package com.mainstreetcode.teammate.model.enums;

import com.google.gson.JsonObject;

public class BlockReason extends MetaData {

    BlockReason(String code, String name) {
        super(code, name);
    }

    public static BlockReason empty() {
        return new BlockReason("", "");
    }

    public static class GsonAdapter extends MetaData.GsonAdapter<BlockReason> {
        @Override
        BlockReason with(String code, String name, JsonObject body) {
            return new BlockReason(code, name);
        }
    }
}

File: app/src/main/java/com/mainstreetcode/teammate/model/enums/BlockReason.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Adetunji Dahunsi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.mainstreetcode.teammate.model.enums

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonObject

class BlockReason internal constructor(code: String, name: String) : MetaData(code, name) {

    class GsonAdapter : MetaData.GsonAdapter<BlockReason>() {
        override fun fromJson(code: String, name: String, body: JsonObject, context: JsonDeserializationContext): BlockReason =
                BlockReason(code, name)
    }

    companion object {

        fun empty(): BlockReason = BlockReason("", "")
    }
}
